rs/nns/governance/src/governance.rs

```diff
@@ -2391,22 +2403,20 @@ impl Governance {
             return Err(GovernanceError::new(ErrorType::NotAuthorized));
         }
 
-        if split_amount_e8s < min_stake + transaction_fee_e8s {
+        let min_split_amount = min_stake.saturating_add(transaction_fee_e8s);
+        if split.amount_e8s < min_split_amount {
             return Err(GovernanceError::new_with_message(
                 ErrorType::InsufficientFunds,
                 format!(
                     "Trying to split a neuron with argument {} e8s. This is too little: \
                       at the minimum, one needs the minimum neuron stake, which is {} e8s, \
                       plus the transaction fee, which is {}. Hence the minimum split amount is {}.",
-                    split_amount_e8s,
-                    min_stake,
-                    transaction_fee_e8s,
-                    min_stake + transaction_fee_e8s
+                    split.amount_e8s, min_stake, transaction_fee_e8s, min_split_amount
                 ),
             ));
         }
 
-        if minted_stake_e8s < min_stake + split_amount_e8s {
+        if minted_stake_e8s < min_stake.saturating_add(split.amount_e8s) {
```

Comment:
See, here is a rare example of where I would say, "yes, we actually want saturation on overflow". What's unusual here is that we do not actually care about the exact sum. All we care about is when the "true" sum exceeds minted_stake_e8s.